### ISSUE TITLE
Rule Proposal: vue/no-mutating-computed

### ISSUE THREAD

**Issue by ThomasKientz** [vuejs/eslint-plugin-vue#2413]
**Please describe what the rule should do:**
Prevent mutating a computed without setter.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- ✗ BAD -->
<script>
export default {
  data: () => ({
    msg: 'foo'
  }),
  computed: {
    foo() {
      return msg
    }
  },
  methods: {
    test() {
      this.foo = 'bar' // <----- no-mutating-computed
    }
  }
}
</script>
```

```vue

<!-- ✓ GOOD -->
<script>
export default {
  data: () => ({
    msg: 'foo'
  }),
  computed: {
    foo:{
      get() {
        return this.msg
      },
      set(newValue) {
       this.msg = newValue
    }
  },
  methods: {
    test() {
      this.foo = 'bar'
    }
  }
}
</script>
```
